docs/cli/ilab-config-current-state.md

```diff
@@ -0,0 +1,90 @@
+# `ilab` Configuration Design as of 0.18.0
+
+## Config Terminology
+
+### Config Layers
+
+There are several locations where config can be specified. These are defined below in order of ascending precedence:
+
+1. The `Config` class in `configuration.py` stores all the general config defaults in code
+1. `config.yaml` is where the user defines their own set of defaults for the system. This can be generated using `ilab config init` and modified using `ilab config edit`. Users can also provide partial config files, for which the rest of the defaults will resort to those defined in the `Config` class.
+1. The user can specify CLI flags when running an `ilab` subcommand, which will override all other configuration.
+
+### Click-isms
+
+[Click](https://click.palletsprojects.com/en/8.1.x/) is a Python library that provides various utilities for Command-Line Interfaces. `ilab` uses these to varying extents to store or pass around config state:
+
+- **Groups & Commands**: Click exposes functionality for 'commands' and 'command groups'. For example, `ilab`, `data`, and `model` are all command groups, whereas `generate`, `train`, and `evaluate` are all commands. We will refer to the latter as subcommands, to clarify that they lie below `ilab` in the command hierarchy.
+
+- **Function Decorators**: CLI commands and subcommands can define their command line arguments in function decorators. Among other things, argument defaults can be defined here. For example:
+
+```python
+@click.option(
+    "--num-instructions",
+    type=click.INT,
+    default=-1,
+    hidden=True,
+)
+```
+
+- **Context**: Click provides a context (`ctx`) object that can be passed across command groups and their members. We use it mainly to pass around the configuration we generate from reading the defaults defined in code and/or the config file (as previously described). Specifically, we assign the configuration in various areas under `ctx.obj`. We elaborate on this later in the doc
+
+- **Click Extensions** `clickext.py` defines some helpful extensions to click functionality that we use throughout the config process. Specifically, `clickext` manages the funnelling of `config` defaults into `click` options. The `clickext.ConfigOptions` class maps values stored in `ctx.obj.config` and `ctx.obj.default_map` to the click arguments defined in a subcommand decorator, for example:
+
+```python
+@click.option(
+    "--data-path",
+    ...
+    cls=clickext.ConfigOption,
+    typ
+)
+```
+
+This allows the function to read the values stored in the context object to fill that variable.
+
+### Additional Args
+
+A subset of the `ilab model train` config variables were deemed more advanced and less likely to be tweaked, so they are exported to a separate section of the config names "additional_args". For example, on linux, the training additional args are stored in `~/.local/share/instructlab/internal/train_configuration/additional/additional_args.yaml`.
```

Comment:
`names` -> `named`

Is it `additional_args` as written here or `additional_arguments` as written below?